plugins/plugin-kubectl/src/controller/client/direct/url.ts

```diff
@@ -32,8 +34,9 @@ function apiOnPathFor(version: string) {
   return version === 'v1' ? 'api/v1' : `apis/${encodeURIComponent(version)}`
 }
 
-function kindOnPathFor(kind: string) {
-  return `/${encodeURIComponent(kind.toLowerCase() + (/s$/.test(kind) ? '' : 's'))}`
+function kindOnPathFor(_kind: string) {
+  const kind = _kind.toLowerCase()
+  return `/${encodeURIComponent(isPlural(kind) ? kind : plural(kind))}`
```

Comment:
I want to be sure I am understanding why the plural check is being used in this way.  (And that probably means this code needs some comments for future maintainability.) Since this was related to the case of "get ingress" failing, I assume this error would have occurred for any k8s resource type that ends in "s".  Is line 39 just ensuring that any resource name that ends in "s" will keep its "s"?

Reply:
When talking to the kubernetes api, the kind-on-path has to be the plural form of the kind, in lowercase; so "pods", not "pod" or "Pod" or "Pods" or even "po".

In the before code, we attempted to meet this constraint with the `/s$/.test()` check. This does not work for kinds that naturally end in an s. Hence, the PR employs the `pluralize` npm to do this in a better fashion.

Reply:
OK, then I think we should add a comment explaining that in the code as that level of detail is not obvious unless someone has debugged the prior issue and that this is sort of a work-around because, as I understand it, this change is not making "ingress" (a singular k8s resource type) into "ingresses" like you describe for making "pod" into "pods" for the API.  The change is more to ensure the resource type name ends in an "apparent plural form" to satisfy the k8s API, even if the resulting name is not truly plural.  This type of subtlety in the code is the kind of thing that benefits from a concise comment if there is ever any other bug that comes up in this area of the code.  (If I am still misunderstanding this, then that is even more reason for a comment to clarify what the code is doing.)

Reply:
it is actually ingresses. this is the kube api. i don't know why they went with the plural form for the api path, kinda weird. i agree with the benefits of a comment.

Reply:
@mark-nc comments added. Thanks!

Reply:
Thanks!  That helps clear it up nicely.